app/tile/menu-button.tsx

```diff
@@ -0,0 +1,62 @@
+import Icon from "app/core/Icon"
+import React from "react"
+import {useDispatch} from "react-redux"
+import {submitSearch} from "src/js/flows/submitSearch/mod"
+import {removeHeadProc} from "src/js/lib/Program"
+import {showContextMenu} from "src/js/lib/System"
+import Layout from "src/js/state/Layout"
+import SearchBar from "src/js/state/SearchBar"
+import styled from "styled-components"
+
+const BG = styled.div`
+  position: relative;
+  padding: 6px;
+  svg {
+    width: 12px;
+    height: 12px;
+    z-index: 1;
+    fill: var(--slate);
+  }
+  &:before {
+    content: "";
+    background: var(--hover-light-bg);
+    position: absolute;
+    top: 0;
+    left: 0;
+    width: 100%;
+    height: 100%;
+    border-radius: 50%;
+    transform: scale(0);
+    transition: transform 100ms;
+  }
+  &:hover:before {
+    transform: scale(1);
+  }
+`
+
+const showMenu = (query) => (dispatch) => {
+  showContextMenu([
+    {
+      label: "Show all records",
+      click() {
+        dispatch(SearchBar.changeSearchBarInput(removeHeadProc(query)))
+        dispatch(Layout.setMainView("search"))
+        dispatch(submitSearch())
+      }
+    }
+  ])
+}
+
+type Props = {
+  query: string
+}
+
+export default function MenuButton({query}: Props) {
+  const dispatch = useDispatch()
+
+  return (
+    <BG onClick={() => dispatch(showMenu(query))} className="menu-button">
```

Comment:
darn, wish we could just use styled components here instead of both 💅 and the css from ".menu-button", makes it a bit difficult to know what the reconciled style will be after the latter overwrites the former

Reply:
You know, the only reason I needed the class name is because I couldn't figure out how to reference this menu-button component inside of the tile.tsx. 

I tried:

const StyledMenuButton = styled(MenuButton)

const BG /* tile bg */ = `
  ${StyledMenuButton} {
     // my styles to show/hide
  }
` 

But that didn't work. Was I doing something wrong?

Reply:
Oh, yeah I think its bc the `styled` template requires the back ticks to work, so:

const StyledMenuButton = styled(MenuButton)``

and then styled components require that the component it wraps takes and passes in a `className` prop. So perhaps more simply, you can just export the menu-button as a styled component like this